internal/usecases/get_cluster_credentials.go

```diff
@@ -129,6 +123,49 @@ func (u *getClusterCredentialsUseCase) run(ctx context.Context) error {
 	return nil
 }
 
+// getAllClustersAuthInformation gets a token per cluster to mimik cross cluster login (for now)
+func (u *getClusterCredentialsUseCase) getAllClustersAuthInformation(ctx context.Context) {
+	wg := &sync.WaitGroup{}
+	m8Clusters, _ := u.clusterServiceClient.GetAll(ctx, &api.GetAllRequest{IncludeDeleted: false})
+	for {
+		m8Cluster, err := m8Clusters.Recv()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			continue
+		}
+
+		clusterAuthInfo := u.config.GetClusterAuthInformation(m8Cluster.Name, u.config.AuthInformation.Username, u.clusterRole)
+		if clusterAuthInfo != nil && clusterAuthInfo.IsValidExact() {
+			continue
+		}
+
+		go func() {
+			wg.Add(1)
+			defer wg.Done()
+			_, _ = u.requestClusterAuthInformation(ctx, m8Cluster)
+		}()
+	}
+	wg.Wait()
+}
+
+func (u *getClusterCredentialsUseCase) requestClusterAuthInformation(ctx context.Context, m8cluster *projections.Cluster) (response *apiGateway.ClusterAuthTokenResponse, err error) {
+	// Get token from gateway
+	response, err = u.clusterAuthClient.GetAuthToken(ctx, &apiGateway.ClusterAuthTokenRequest{
+		ClusterId: m8cluster.Id,
+		Role:      u.clusterRole,
+	})
+	if err != nil {
+		return
+	}
+
+	// Cache credentials
+	u.config.SetClusterAuthInformation(m8cluster.Name, u.config.AuthInformation.Username, u.clusterRole, response.AccessToken, response.Expiry.AsTime())
```

Comment:
Doing this concurrently could lead to the config being broken. The underlying stuff is not thread safe I guess so you would need a lock around this.

Reply:
You are right. I forgot that I'm caching here as well (one of the reasons why a method should have one job only :D).